caikit/runtime/service_generation/protoable.py

```diff
@@ -91,6 +107,47 @@ def handle_protoables_in_union(arg_type: Type) -> Type:
     log.debug("Skipping non-protoable argument type [%s]", arg_type)
 
 
+def _make_union_list_source_type_name(args: List) -> str:
+    """Make the name for  source class that wraps the given types"""
+    return f"UnionList{''.join([get_args(arg)[0].__name__.capitalize() for arg in args])}Source"
+
+
+def make_union_dataobjects(union_protoables: List) -> Type[DataBase]:
```

Comment:
I think this is introducing one unnecessary level of indirection. If I'm reading this correctly, this is creating a net-new message with a single field `union_list` whose type hint is a union of the associated sequence wrapper classes. I think that intermediate class is not needed since the message types in the `Union` already provide the wrapping to allow them to exist in a `Union`. Basically, I think we want to take the type hint you're setting on line 137 and use that as the type hint for the actual field that's being auto-deduced, rather than adding the indirection of the `union_list` field.

Reply:
Oh or you want to return the `Union[tuple(data_object_list)]` as is which becomes the type of the `Union List`? 🤔 Hmm interesting